src/main/java/software/amazon/smithy/lsp/Main.java

```diff
@@ -61,21 +65,28 @@ public static void main(String[] args) {
         Socket socket = null;
         InputStream in;
         OutputStream out;
-
+        List<String> argList = Arrays.asList(args);
         try {
-            String port = args[0];
+            Optional<Exception> launchFailure;
+            String port = getOrDefault(argList, 0, "0");
+            String type = getOrDefault(argList, 1, null);
             // If port is set to "0", use System.in/System.out.
             if (port.equals("0")) {
                 in = System.in;
                 out = System.out;
+                launchFailure = launch(in, out);
+            } else if (type != null && type.equals("--ws")) {
```

Comment:
```suggestion
            } else if ("--ws".equals(type)) {
```
Minor, don't have to do the null check like this.